lib/node_modules/@stdlib/fs/resolve-parent-paths/lib/async.js

```diff
@@ -0,0 +1,425 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var resolve = require( 'path' ).resolve;
+var isStringArray = require( '@stdlib/assert/is-string-array' ).primitives;
+var isFunction = require( '@stdlib/assert/is-function' );
+var cwd = require( '@stdlib/process/cwd' );
+var exists = require( '@stdlib/fs/exists' );
+var filled = require( '@stdlib/array/base/filled' );
+var format = require( '@stdlib/string/format' );
+var validate = require( './validate.js' );
+
+
+// VARIABLES //
+
+var MODES = {
+	'first': first,
+	'some': some,
+	'all': all,
+	'each': each
+};
+
+
+// FUNCTIONS //
+
+/**
+* Asynchronously resolves the first path match from a set of paths by walking parent directories.
+*
+* @private
+* @param {Array<string>} paths - paths to resolve
+* @param {string} dir - base directory
+* @param {Callback} done - callback to invoke after resolving paths
+* @returns {void}
+*/
+function first( paths, dir, done ) {
+	var child;
+	var spath;
+	var FLG = 0; // flag to track if we are done traversing a directory level
+	var out;
+	var i;
+
+	// Start at a base directory and continue moving up through each parent directory...
+	out = [];
+	for ( i = 0; i < paths.length; i++ ) {
+		spath = resolve( dir, paths[ i ] );
+		exists( spath, getCallback( spath ) );
+	}
```

Comment:
I don't think we should do this. Otherwise, we're blasting the filesystem with `access` requests, especially when only one request returns the file and the rest will be wasted. I suggest doing one async request at a time. You can find examples of this pattern elsewhere in the repository using `next` callbacks. Example: https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/utils/async/series-waterfall/lib/factory.js